File: src/main/java/dominio/metodologias/CondicionPrioritaria.java
package dominio.metodologias;

import dominio.empresas.Empresa;

import javax.persistence.Entity;
import javax.persistence.Table;

@Entity
@Table(name = "condiciones_prioritarias")
public class CondicionPrioritaria extends Condicion{
	
	private CondicionPrioritaria() {} //Necesario para persistir la clase
	
	public CondicionPrioritaria(OperandoCondicion operando, OperacionRelacional operacionRelacional) {
		this.operando = operando;
		this.operacionRelacional = operacionRelacional;
	}
	
	public boolean esMejorQue(Empresa empresa1, Empresa empresa2){
		return operacionRelacional.aplicarA(operando.valorPara(empresa1), operando.valorPara(empresa2));
	}
	
}

File: src/main/java/dondeInvierto/dominio/metodologias/CondicionPrioritaria.kt
package dondeInvierto.dominio.metodologias

import dondeInvierto.dominio.empresas.Empresa

import javax.persistence.Entity
import javax.persistence.Table

@Entity
@Table(name = "condiciones_prioritarias")
class CondicionPrioritaria(operando: OperandoCondicion, operacionRelacional: OperacionRelacional) : Condicion(operando, operacionRelacional) {

    fun esMejorQue(empresa1: Empresa, empresa2: Empresa, anioActual: Int): Boolean {
        return operacionRelacional.aplicarA(operando.valorPara(empresa1, anioActual), operando.valorPara(empresa2, anioActual))
    }

}
